prometheus/doc/README.md

```diff
@@ -0,0 +1,84 @@
+# Goal
+
+Monitoring all compoments in pai, provide insight on detectiving system/hardware failuring and
+analysing jobs performance.
+
+To accomplish the goal, this module should be:
+* Robust: monitoring should try its best to not crash. Other compoments/people relay on monitoring system to find root causes or trigger alert.
+* Minimize performance impact.
+* Minimize latency between metrics generation and metrics consumption.
+
+# Architecture
+
+![Architecture](architecture.png)
+
+We have three parts consisting Pai's monitoring system: `watchdog`, `exporter` and
+[`prometheus`](https://prometheus.io/).
+
+`prometheus` is an open source solution for metrics collecting, storage and querying.
+
+`watchdog` is a single instance pod responsible for monitoring k8s/nodes/pods health, it will first
+list all pod/node from kubernetes api server and check their status and log their status, this is very
+helpful for debugging.
+
+`exporter` are those pods running in the lefe side of node, they are responsible for collecting
+metrics from jobs/nodes/gpus. There are two containers running inside `exporter` pod: `gpu_exporter`
+and [`node_exporter`](https://github.com/prometheus/node_exporter): `gpu_exporter` exposes job/gpu
+metrics to volume mounted in `/datastorage/prometheus`.
+
+Metrics generated by `watchdog` and `gpu_exporter` are collected by `node_exporter` container running
+inside `exporter` pod. Those metrics are scraped by `node_exporter` container. `node_exporter` also
+expose node metricss like node cpu/memory/disk usage.
+
+# Metrics collected
+
+Some important metrics are listed below.
```

Comment:
as will add more and more metrics, how about make this as a separate md file and add that file's link here